Feature/PJ_SIM_SUNBNB_2023-17-[FE] Tab Component

RonnieRII opened the pull request:
# Issue Link
https://framgiaph.backlog.com/view/PJ_SIM_SUNBNB_2023-17

# Definition of Done
* Create a tab component for hosts with the components:
    * [x] Your Listings
    * [x] Your Reservations
    
* Create a tab component for adding a listing by hosts with the components:
    * [x] Listing
    * [x] Pricing
    * [x] Description
    * [x] Photos
    * [x] Amenities
    * [x] Publish Button
    
# Note
Created two pages for the first tab (HostTab) and second tab (ListingTab) as described above

# Screenshots/Recordings
![image](https://github.com/framgia/pj_sim-sunbnb/assets/138445425/e75c7f7a-a33d-4ed7-b01a-350b70eadb0a)
![image](https://github.com/framgia/pj_sim-sunbnb/assets/138445425/f739a342-3231-4635-811b-70b8f51c4d45)

LennardUy104 commented on fe/src/components/ListingTab.vue:
```diff
@@ -0,0 +1,76 @@
+        <!--Publish-->
+        <v-list class="d-flex justify-center pt-5 pl-2 pr-2">
+            <v-btn class="flex-grow-1" color="red" rounded="lg" id="btn" w>
+                Publish
```
I think this button is not necessary since there will be a save button on the add new listing page
thanks

RonnieRII replied:
Noted, thank you!

LennardUy104 commented on package-lock.json:
What is this for?

RonnieRII replied:
I think it was generated when I initialized the project. I have deleted it as it has no effect on the code itself. Thank you!

LennardUy104 reviewed (approved):
lgtm! thanks!!

RonnieRII committed:
PJ_SIM_SUNBNB_2023-17-FE-Tab-Component

PJ_SIM_SUNBNB_2023-17-FE-Tab-Component

Added to AllComponents and functions to the tabs

fix route index.js

resolve changes

LennardUy104 reviewed (commented):
Please squash Thanks

roseaugusto commented on fe/src/components/Tab/HostTab.vue:
```diff
@@ -0,0 +1,38 @@
+        <v-list-item-group>
+
+          <!--Listings-->
+          <v-list-item href="#listings" @click="activeOption = 'l'" :class="{'option-active' : activeOption === 'l'}">
```
Since `activeOption` is just 2 options we can change the type to boolean, please change the naming too to `isActive`.

```suggestion
          <v-list-item href="#listings" @click="function instead" :class="{'option-active' : activeOption}">
```

RonnieRII replied:
Just did this ma'am 🙇

roseaugusto commented on fe/src/components/Tab/HostTab.vue:
```diff
@@ -0,0 +1,38 @@
+          </v-list-item>
+
+          <!--Reservations-->
+          <v-list-item href="#reservations" @click="activeOption = 'r'" :class="{'option-active' : activeOption === 'r'}">
```
same here

RonnieRII replied:
Just did this also ma'am 🙇

roseaugusto commented on fe/src/components/Tab/ListingTab.vue:
```diff
@@ -0,0 +1,67 @@
+          <v-list-item-group>
+  
+            <!--Listing-->
+            <v-list-item href="#listing" @click="activeOption = 'li'" :class="{'option-active' : activeOption === 'li'}">
```
create a function wherein you will check in switch case form the active option/tab and just call the function in class

RonnieRII replied:
Just did this ma'am 🙇

roseaugusto commented on fe/src/pages/AllComponents.vue:
```diff
@@ -5,6 +5,12 @@
     <ReviewForm />
     <ReviewRoom />
     <ReviewHost />
+<<<<<<< Updated upstream
```
please fix this

RonnieRII replied:
Just fixed this ma'am 🙇

roseaugusto commented on fe/src/pages/AllComponents.vue:
```diff
@@ -14,6 +20,12 @@ import HomepageBanner from "../components/HomepageBanner.vue";
 import ReviewForm from "../components/Review/ReviewForm.vue";
 import ReviewRoom from "../components/Review/ReviewRoom.vue";
 import ReviewHost from "../components/Review/ReviewHost.vue";
+<<<<<<< Updated upstream
```
this one as well

RonnieRII replied:
Just fixed this as well ma'am 🙇

LennardUy104 commented:
Resolve all conflict

roseaugusto commented on fe/src/components/Tab/HostTab.vue:
```diff
@@ -0,0 +1,46 @@
+  let isActive = true
+
+  function changeActive(option) {
+    if(option === 'list') {
```
use switch case here instead

LennardUy104 replied:
noted maam

roseaugusto commented on fe/src/pages/HelloWorld.vue:
```diff
@@ -3,17 +3,30 @@
   <ReviewForm />
   <ReviewRoom />
   <ReviewHost />
+<<<<<<< Updated upstream
```
there are still conflicts, please fix

LennardUy104 replied:
sorry i forgot to remove this file. will remove this file since I remove this file on the other PR's maam since there is already a file that contains all components

roseaugusto commented on fe/src/router/index.js:
```diff
@@ -1,18 +1,18 @@
-    path: "/registration",
-    component: Registration,
-  },
+//  {
```
remove this if not needed

LennardUy104 replied:
okay maam
